cargo-miri/bin.rs

```diff
@@ -510,8 +567,30 @@ fn phase_cargo_miri(mut args: env::Args) {
         &host
     };
 
-    // Forward all further arguments to cargo.
-    cmd.args(args);
+    let mut target_dir = None;
+
+    // Forward all arguments before `--` other than `--target-dir` and its value to Cargo.
+    for arg in ArgSplitFlagValue::new(&mut args, "--target-dir") {
+        match arg {
+            Ok(value) => {
+                if target_dir.is_some() {
+                    show_error(format!("`--target-dir` is provided more than once"));
+                }
+                target_dir = Some(value.into());
+            }
+            Err(arg) => drop(cmd.arg(arg)),
```

Comment:
What are you dropping here?

Reply:
I'm using `drop` to make the two match arms to have the same type (`&mut Command` -> `()`).

I think this can also be written as:
```suggestion
            Err(arg) => {
                cmd.arg(arg));
            }
```
I don't have a preference here. :shrug:

Reply:
Yes please use a block and `;` over `drop`.